src/LibAwsCommon.jl

```diff
@@ -43,7 +43,23 @@ for name in names(@__MODULE__; all=true)
     @eval export $name
 end
 
-function init(allocator=aws_default_allocator())
+const DEFAULT_AWS_ALLOCATOR = Ref{Ptr{aws_allocator}}(C_NULL)
+
+function default_aws_allocator()
+    if DEFAULT_AWS_ALLOCATOR[] == C_NULL
+        DEFAULT_AWS_ALLOCATOR[] = aws_default_allocator()
+    end
```

Comment:
needs to be protected by a lock, and should also use `set_default_aws_allocator!` if we're going to have that setter.